library/psa_crypto.c

```diff
@@ -4413,7 +4564,9 @@ static psa_status_t psa_key_derivation_tls12_prf_setup(
 
     return( PSA_SUCCESS );
 }
+#endif /* PSA_PRE_1_0_KEY_DERIVATION */
 
+#if defined(PSA_PRE_1_0_KEY_DERIVATION)
```

Comment:
These guard and the previous #endif seems a bit redundant given that both code blocks have the same conditional macro checks.

Reply:
This redundancy is intentional, makes it easier to remove/refactor these functions individually.